README.md
## Module OGSpy Flottes

This README would normally document whatever steps are necessary to get your application up and running.

### What is this repository for? ###

* Quick summary
* Version
* [Learn Markdown](https://bitbucket.org/tutorials/markdowndemo)

### How do I get set up? ###

* Summary of set up
* Configuration
* Dependencies
* Database configuration
* How to run tests
* Deployment instructions

### Contribution guidelines ###

* Writing tests
* Code review
* Other guidelines

### Who do I talk to? ###

* Repo owner or admin
* Other community or team contact

ChangeLog.txt
Version 1.10
- mise à niveau pour Ogspy 3.0.7
- correction des index de boucle sur les id de planètes et lunes

Version 1.08c
- typo dans certains commentaires

Version 1.08b
- correction footer
- correction xtense2 si pas installe
- changement du mode de recuperation de la version

Version 1.08a
- correction code xtense2
- compatibilité totale avec xtense2
- ajout banniere du mod
- modification du menu
- corrections code mineures

Version 1.07
- Correction de la prise en compte du séparateur de milliers
- Correction d'un Warning en cas de division par zéro dans l'export bbcode
- Ajout de la liaison avec Xtense V2

Version 1.06
- Ajout export MyBB
- Ajout conversion/preview pour MyBB
- Correction inversion Ion et Gauss export phpbb et phpbb detaille
- Correction pourcentage de protection de la flotte par la defense
- Changement mise en forme des exportations

de V 1.01 à V 1.05
- correction affichage mines et usines en page Empire
- Mise à jour en 1 passe
- Correction du bug de la page Empire sous OGSpy.
- Correction du bug d'adressage du fichier ogame.php ou univers.php en fonction des paramètres admin.
- Correction du bug Import de flotte.

V 1.00
- Suppression simulation empire.
- Décalage des planètes en cas ne nouvelles colonies.
- Révision des exports BBCode.
- ajout de 2 champs en page admin pour le type de serveur et le nombre de planète à afficher par page.
Support OGSpy
- Support V 3.02C  OK
- Support 3.03 (non testé)
- Support 3.10 (Non testé)
Support UniSpy
- Support 1.00 UNISPY OK

V 0.76
- Possibilité de désactiver le partage des données Empire aux autres membres de votre alliance.
Les flottes sont partagé aux membres de votre alliance et aux joueurs que vous rajoutez dans la liste

V 0.75
- Ajout du traqueur dans la liste des vaisseaux

V 0.73 et 0.74
- Ajout d'un export BBCode plus détaillé (origine @Moa666).
- Corrections mineures (Orthographe et autres...)
- Correction des couts matières par vaisseau.


V 0.72
- Correction du code PHP (sur insistance de oXid_FoX).
- Corrections mineures (Orthographe et autres...)

V 0.71
- Correction install.php
- Corrections mineures (Orthographe et autres...)

V 0.7
- Intégration de la page insertion de @Lithie (avec quelques modifications). Possibilité de sélectionner une planète par son nom ou ses coordonnées.
- Ajout de la gestion des Satellites Solaires.
- Révision de l'export BBCode (joueur non classé ou classement absent, total points)

V 0.6
- Intégration des écrans du mod Empire avec la gestion de groupe. Ceci n'est vraiment utile que pour les serveurs multi-alliances.

V 0.5
- gestion des droits par alliance (ou groupe de joueur)
Les droits se font par groupe d'utilisateurs (flottes_nom_alliance) - administration OGSpy
exemple flottes_Monalliance
Vous pouvez supprimer le groupe mod_flotte

V 0.4
- Prévisualisation de l'export BBCode.
- Epuration de code, et refonte pour une utilisation plus souple.
- Prise en compte du nouveau vaisseau.


V 0.3
- Correction de bugs pour désinstaller, mise à jour.
- Finalisation fonction import auto via plug-in Naq version > 1.3.0

V 0.21
- Ajout de l'affichage des défenses des autres joueurs.

V 0.2
- Gestion des couleurs (les mêmes pour tous le monde), dans OGSpy et sur le forum.

V 0.1c
- normalement identique à 0.1b mais comme je n'été pas sûr!!!

V 0.1b
- Rajout de test
- Correction de bug date (j'espère)
- Protection contre les erreurs d'import de lune.

v0.1:
- Première version.



update.php
<?php
/***************************************************************************
*   filename    : update.php
*   desc.       : 1.08beta1
*   Author      : 
*   created     : 
*   modified    : AirBAT
*   last modif. : added upgrade from v1.07  
***************************************************************************/

if (!defined('IN_SPYOGAME')) {
    die("Hacking attempt");
}

global $db,$table_prefix;

//define("TABLE_MOD_FLOTTES_ADM", substr(TABLE_USER, 0, strlen(TABLE_USER)-4)."mod_flottes_admin");
define("TABLE_MOD_FLOTTES_ADM", $table_prefix."mod_flottes_admin");
define("TABLE_GROUP", $table_prefix."group");
define("TABLE_MOD_FLOTTES", $table_prefix."mod_flottes");
define("FLOTTES_FOLDER","mod/flottes");

    	
// Verification version
if (file_exists(FLOTTES_FOLDER.'/version.txt')) {
    list($mod_name,$version) = file(FLOTTES_FOLDER.'/version.txt'); 
   
	
}
else {
    die('Le fichier "version.txt" est introuvable !');
}

$query = "SELECT `id`, `version` FROM ".TABLE_MOD." WHERE `action`='flottes'";
$result = $db->sql_query($query);

		
if(list($mod_id, $version) = $db->sql_fetch_row($result)) {
	SWITCH($version) {

		CASE "0.2" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.3'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		// echo "0.20";
		// break;
		
		CASE "0.21" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.3'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.3";
		// echo "0.21";
		// break;
		
		CASE "0.3"  :
		$query = "UPDATE ".TABLE_MOD." SET version='0.6'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$query="ALTER TABLE ".TABLE_MOD_FLOTTES." ADD `TRA` INT NOT NULL DEFAULT '0'";
		$db->sql_query($query);
		$version="0.6";
		// echo "0.3";
		// break;
		
		CASE "0.4"  :
		$query = "UPDATE ".TABLE_MOD." SET version='0.6'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.6";
		// echo "0.4";
		// break;
		
		CASE "0.5"  :
		$query = "UPDATE ".TABLE_MOD." SET version='0.6'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.6";
		// echo "0.5";
		// break;
		
		CASE "0.6" :
		$query ="ALTER TABLE ".TABLE_MOD_FLOTTES." ADD `SAT` INT NOT NULL DEFAULT '0'";
		$db->sql_query($query);
		$query = "UPDATE ".TABLE_MOD." SET version='0.7'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.7";
		// echo "0.6";
		// break;
		
		CASE "0.7" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.7";
		// break;
		
		CASE "0.71" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.71";
		// break;
		
		CASE "0.72" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.72";
		// break;
		
		CASE "0.73" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.73";
		// break;
		
		CASE "0.74" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.74";
		// break;
		
		CASE "0.75" :
		$query = "UPDATE ".TABLE_MOD." SET version='0.76'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version="0.76";
		// echo "0.75";
		// break;
		
		CASE "0.76" :
		$query = "ALTER TABLE  ".TABLE_MOD_FLOTTES_ADM." ADD `GAME` VARCHAR( 8 ) NOT NULL DEFAULT 'OGAME' ";
		$db->sql_query($query);
		
		$query = "ALTER TABLE  ".TABLE_MOD_FLOTTES_ADM." ADD `nbpla` TINYINT NOT NULL DEFAULT '9' ";
		$db->sql_query($query);
		
		$query = "ALTER TABLE ".TABLE_MOD_FLOTTES." DROP PRIMARY KEY, ADD PRIMARY KEY( `user_id`,  `planet_id`,  `planet_name`,  `coordinates`)";
		$db->sql_query($query);
			
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "0.76";
		// break;
		
		CASE "1.00" :
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "1.00";
		// break;
		
		CASE "1.01" :
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "1.01";
		// break;
		
		CASE "1.02" :
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "1.02";
		// break;
		
		CASE "1.03" :
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "1.03";
		// break;

		CASE "1.04" :
		$query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
		$db->sql_query($query);
		$version=$vfinale;
		// echo "1.04";
		break;

        CASE "1.05" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.05";
        break;
        		
        CASE "1.06" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.06";
        break;
                
        CASE "1.07" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.07";
        break;
                
        CASE "1.08a" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.08a";
        break;
                
        CASE "1.08b" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.08b";
        break;
		
        CASE "1.10" :
        $query = "UPDATE ".TABLE_MOD." SET version='".$vfinale."'  WHERE action='mod_flottes' AND version='".$version."'";
        $db->sql_query($query);
        $version=$vfinale;
        // echo "1.10";
        break;
		
        default:
		$mod_folder = "flottes";
		$mod_name = "flottes";
		update_mod($mod_folder, $mod_name);

// / Nouvelle table pour la gestion des couleurs
		// $query = "CREATE TABLE ".TABLE_MOD_FLOTTES_ADM." (".
			// " group_name varchar(20) not null default '',".
			// " color_fleet varchar(8) not null default '',".
			// " color_fleet_old varchar(8) not null default '',".
			// " color_fleet_user varchar(8) not null default '',".
			// " color_fleet_point varchar(8) not null default '',".
			// "color_fleet_alli varchar(8) not null default '',".
			// "color_bbc_1 varchar(8) not null default '',".
			// "color_bbc_2 varchar(8) not null default '',".
			// "color_bbc_3 varchar(8) not null default '',".
			// "color_bbc_4 varchar(8) not null default '',".
			// "color_bbc_5 varchar(8) not null default '',".
			// " primary key (group_name))";
		// $db->sql_query($query);

		// $request = "INSERT INTO ".TABLE_MOD_FLOTTES_ADM." (group_name,color_fleet,color_fleet_old,color_fleet_user, color_fleet_point, color_fleet_alli, ";
		// $request .="color_bbc_1, color_bbc_2, color_bbc_3, color_bbc_4, color_bbc_5)   VALUES ('mod_flottes', 'lime', 'yellow', 'red', 'lime', 'teal', ";
		// $request .="'yellow', 'red', 'green', 'cyan', 'orange')";
		// $db->sql_query($request);
		
		// break;

	}

}
/*
		$request = "Select group_name from ".TABLE_GROUP." where group_name='mod_flottes'";
		$result=$db->sql_query($request);
		$group=mysql_num_rows($result);
		echo $group." - ".$version;
		
		if ($group==0) {
		
		$request = "INSERT INTO ".TABLE_GROUP." (group_id, group_name)   VALUES ('','mod_flottes')";
		$db->sql_query($request);}
*/		

// Enregistrement du Mod Flottes dans Xtense2
define("TABLE_XTENSE_CALLBACKS", $table_prefix."xtense_callbacks");

// On regarde si la table xtense_callbacks existe
$query = "SELECT id, version FROM ".TABLE_MOD." WHERE action='flottes'";
$resultid = $db->sql_query($query);
list($mod_id, $version) = $db->sql_fetch_row($resultid);
$result = $db->sql_query('show tables from '.$db->dbname.' like "'.TABLE_XTENSE_CALLBACKS.'" ');
if($db->sql_numrows($result) != 0){

	//On regarde si le mod Flottes est inscrit dans xtense2
	$result = $db->sql_query("Select * From ".TABLE_XTENSE_CALLBACKS." where mod_id = $mod_id");

	// s'il n'y est pas, on l'ajoute
	if($db->sql_numrows($result) == 0)
		$db->sql_query("INSERT INTO ".TABLE_XTENSE_CALLBACKS." (mod_id, function, type, active) VALUES ('$mod_id', 'flottes_import_fleet', 'fleet', 1)");
	echo("<script> alert('La compatibilité du mod Flottes avec le mod Xtense2 \n est installée !') </script>");		
}
else {
	echo("<script> alert('Le mod Xtense 2 n'est pas installé. La compatibilité du mod Flottes ne sera donc pas installé !\nPensez Ã  installer Xtense 2 c'est pratique ;)') </script>");
}
?>

